hack/testing/logging.sh

```diff
@@ -0,0 +1,261 @@
+kind: ServiceAccount
+metadata:
+  name: logging-deployer
+secrets:
+- name: logging-deployer
+' | oc create -f -"
+os::cmd::expect_success "oc policy add-role-to-user edit system:serviceaccount:logging:logging-deployer"
+if [ -n "$USE_LOGGING_DEPLOYER" ] ; then
+    imageprefix="docker.io/openshift/origin-"
+elif [ -n "$USE_LOGGING_DEPLOYER_SCRIPT" ] ; then
+    pushd $OS_O_A_L_DIR/deployment
+    IMAGE_PREFIX="openshift/origin-" PROJECT=logging ./run.sh
+    popd
+    imageprefix=
+else
+    os::cmd::expect_success "oc process \
+       -f $OS_O_A_L_DIR/hack/templates/dev-builds.yaml \
+       -v LOGGING_FORK_URL=$GIT_URL,LOGGING_FORK_BRANCH=$GIT_BRANCH \
+       | oc create -f -"
+    os::cmd::expect_success "wait_for_builds_complete"
+    imageprefix=`oc get is | awk '$1 == "logging-deployment" {print gensub(/^([^/]*\/logging\/).*$/, "\\\1", 1, $2)}'`
+    sleep 5
+fi
+log_priv_user=$(create_valid_file add-privileged-logging-user.yml)
+os::cmd::expect_success "oc get scc/privileged -o yaml > $log_priv_user"
+os::cmd::expect_success "echo '- system:serviceaccount:logging:aggregated-logging-fluentd' >> $log_priv_user"
+os::cmd::expect_success "oc replace -f $log_priv_user"
+sleep 5
+os::cmd::expect_success "oadm policy add-cluster-role-to-user cluster-reader \
+                      system:serviceaccount:logging:aggregated-logging-fluentd"
+sleep 5
+if [ ! -n "$USE_LOGGING_DEPLOYER_SCRIPT" ] ; then
+    os::cmd::expect_success "oc process \
+                          -f $OS_O_A_L_DIR/deployment/deployer.yaml \
+                          -v ENABLE_OPS_CLUSTER=$ENABLE_OPS_CLUSTER,IMAGE_PREFIX=$imageprefix,KIBANA_HOSTNAME=kibana.example.com,ES_CLUSTER_SIZE=1,PUBLIC_MASTER_URL=https://localhost:8443${masterurlhack} \
+                          | oc create -f -"
+    os::cmd::try_until_text "oc describe bc logging-deployment | awk '/^logging-deployment-/ {print \$2}'" "complete"
+    os::cmd::try_until_text "oc get pods -l component=deployer" "Completed" "$(( 3 * TIME_MIN ))"
+fi
+# see if expected pods are running
+os::cmd::try_until_text "oc get pods -l component=es" "Running" "$(( 3 * TIME_MIN ))"
+os::cmd::try_until_text "oc get pods -l component=kibana" "Running" "$(( 3 * TIME_MIN ))"
+os::cmd::try_until_text "oc get pods -l component=curator" "Running" "$(( 3 * TIME_MIN ))"
+if [ "$ENABLE_OPS_CLUSTER" = "true" ] ; then
+    os::cmd::try_until_text "oc get pods -l component=es-ops" "Running" "$(( 3 * TIME_MIN ))"
+    os::cmd::try_until_text "oc get pods -l component=kibana-ops" "Running" "$(( 3 * TIME_MIN ))"
+    os::cmd::try_until_text "oc get pods -l component=curator-ops" "Running" "$(( 3 * TIME_MIN ))"
+fi
+
+# this fails because the imagestreams already exist
+os::cmd::expect_failure_and_text "oc process logging-support-template | oc create -f -" "already exists"
+# scale up a fluentd pod
+os::cmd::try_until_success "oc get dc logging-fluentd"
+os::cmd::expect_success "oc scale dc logging-fluentd --replicas=1"
```

Comment:
label the nodes so the daemon set selects them for placement of fluentd:

    oc label nodes --all logging-infra-fluentd=true